messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+                .getEntityManager()
+                .createQuery(
+                        "SELECT dl.sequence FROM DeadLetterEntry dl where dl.processingGroup=:processingGroup order by dl.lastTouched asc",
+                        String.class)
+                .setParameter("processingGroup", processingGroup)
+                .getResultList();
+
+
+        return () -> {
+            Iterator<String> sequenceIterator = sequences.iterator();
+            return new Iterator<Iterable<DeadLetter<? extends M>>>() {
+                @Override
+                public boolean hasNext() {
+                    return sequenceIterator.hasNext();
+                }
+
+                @Override
+                public Iterable<DeadLetter<? extends M>> next() {
+                    String next = sequenceIterator.next();
+                    return deadLetterSequence(next);
+                }
+            };
+        };
+    }
+
+    /**
+     * Converts a {@link DeadLetterEntry} from the database into a {@link JpaDeadLetter}, using the configured
+     * {@link DeadLetterJpaConverter DeadLetterJpaConverters} to restore the original message from it.
+     *
+     * @param entry The entry to convert.
+     * @return The {@link DeadLetter} result.
+     */
+    @SuppressWarnings("unchecked")
+    private JpaDeadLetter<M> toLetter(DeadLetterEntry entry) {
+        MetaData deserializedDiagnostics = serializer.deserialize(new SimpleSerializedObject<>(
+                entry.getDiagnostics(),
+                byte[].class,
+                MetaData.class.getName(),
+                null));
+        DeadLetterJpaConverter<M> converter = (DeadLetterJpaConverter<M>) converters
+                .stream()
+                .filter(c -> c.canConvert(entry.getMessage()))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("No such converter"));
+        return new JpaDeadLetter<>(entry,
+                                   deserializedDiagnostics,
+                                   converter.fromEntry(entry.getMessage(), serializer));
+    }
+
+    @Override
+    public boolean isFull(@Nonnull Object sequenceIdentifier) {
+        String identifier = toIdentifier(sequenceIdentifier);
+        return maximumNumberOfQueuesReached(identifier) || maximumQueueSizeReached(identifier);
+    }
+
+    @Override
+    public long maxSequences() {
+        return maxSequences;
+    }
+
+    @Override
+    public long maxSequenceSize() {
+        return maxSequenceSize;
+    }
+
+    @Override
+    public boolean process(@Nonnull Predicate<DeadLetter<? extends M>> sequenceFilter,
+                           @Nonnull Function<DeadLetter<? extends M>, EnqueueDecision<M>> processingTask) {
+
+        Optional<JpaDeadLetter<M>> first = findFirstLetterOfEachAvailableSequence()
+                .stream()
+                .map(this::toLetter)
+                .filter(sequenceFilter)
+                .findFirst();
```

Comment:
I think it would be nicer if the `process` task could loop through the available sequences it has found, instead of only retrieving the first one.
This would cover for the scenario that several tasks trigger at once to process dead-letters for the same processing group.

That would make it so that we need to catch the `UnableToClaimDeadLetterException`, and move to the following entry in the list of available sequences.